packages/intl-locale/index.ts

```diff
@@ -695,6 +757,43 @@ export class Locale {
     }
   }
 
+  /**
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getWeekInfo
+   * https://tc39.es/proposal-intl-locale-info/#sec-Intl.Locale.prototype.getWeekInfo
+   */
+  public getWeekInfo() {
+    try {
```

Comment:
not in spec

Reply:
I don't fully understand what you mean by "not in the spec":  in https://tc39.es/proposal-intl-locale-info/#sec-Intl.Locale.prototype.getWeekInfo, it's mentioned `Perform ! CreateDataPropertyOrThrow` which itself says "If success is false, throw a TypeError exception"

<img width="684" alt="image" src="https://github.com/formatjs/formatjs/assets/22725671/2d32dab8-6a50-4fcf-bb68-f2372d43a69d">

I didn't write this original code (as I copy pasted it from `public get weekInfo()`) but if I understand this code properly this `try {} catch {}` is for changing all possible errors into `TypeError` as mentioned in the spec.

Can you help me on how this code should be adapted?

Reply:
for this one you can just remove the `try/catch`. `defineProperty` & `CreateDataPropertyOrThrow` both call `DefinePropertyOrThrow` so that should already be covered

Reply:
Updated 👍